File: AgentenSimulation_1/src/Agent.java
public abstract class Agent {

    private AgentIndexContract round_best;

    private AgentIndexContract global_best;

    public AgentIndexContract getRound_best() {
        return round_best;
    }

    public void setRound_best(AgentIndexContract round_best) {
        this.round_best = round_best;
    }

    public AgentIndexContract getGlobal_best() {
        return global_best;
    }

    public void setGlobal_best(AgentIndexContract global_best) {
        this.global_best = global_best;
    }

    public abstract boolean vote(Contract contract, Contract proposal);

    public abstract AgentIndexContract printUtility(Contract contract);

    public abstract int getContractSize();


    public abstract boolean[] voteLoop(Contract[] contracts, int acceptanceAmount);

    public abstract int voteEnd(Contract[] contracts);

}

File: AgentenSimulation_Rust/src/agent.rs
use std::collections::HashSet;
use rayon::iter::{IntoParallelIterator, IntoParallelRefIterator, ParallelIterator};
use crate::contract::{Contract, CostContract};

pub trait Agent {
    fn get_contract_size(&self) -> usize;
    fn vote_many(&mut self, contracts: &Vec<Contract>, acceptance_amount: usize) -> Vec<bool>;
    fn vote_end(&mut self, contracts: &HashSet<Contract>) -> &Contract;
    fn _eval(&self, solution: &Contract) -> isize;
    fn get_round_best(&self) -> &CostContract;
    fn get_global_best(&self) -> &CostContract;
    fn evaluate_sync(&mut self, contracts: &Vec<Contract>) -> Vec<CostContract>;
    fn evaluate_async<'a, T>(&mut self, contracts: &T) -> Vec<CostContract>
    where T:ParallelIterator<Item=&'a Contract>;
}